parts/kubernetesagentresourcesvmas.t

```diff
@@ -118,18 +118,18 @@
         "[concat('Microsoft.Storage/storageAccounts/',variables('storageAccountPrefixes')[mod(add(add(div(copyIndex(variables('{{.Name}}Offset')),variables('maxVMsPerStorageAccount')),variables('{{.Name}}StorageAccountOffset')),variables('dataStorageAccountPrefixSeed')),variables('storageAccountPrefixesCount'))],variables('storageAccountPrefixes')[div(add(add(div(copyIndex(variables('{{.Name}}Offset')),variables('maxVMsPerStorageAccount')),variables('{{.Name}}StorageAccountOffset')),variables('dataStorageAccountPrefixSeed')),variables('storageAccountPrefixesCount'))],variables('{{.Name}}DataAccountName'))]",
   {{end}}
 {{end}}
-        "[concat('Microsoft.Network/networkInterfaces/', variables('{{.Name}}VMNamePrefix'), 'nic-', copyIndex(variables('{{.Name}}Offset')))]",
+        "[concat('Microsoft.Network/networkInterfaces/', variables('{{.Name}}VMNamePrefix'), 'vm', copyIndex(variables('{{.Name}}Offset')), '-nic')]",
```

Comment:
While changing some these names might be better, changing resource names has a bigger impact that might not be obvious at first glance:
ACS Engine is used by ACS service and RP might have some assumptions about resource naming conventions during scale up/down/upgrade/other scenarios that might break the service if this change is taken.
Updating k8s-agentpool1-kubetest-1 to k8s-kubetest-agentpool1-vm1 increases the length of resource name that's not desirable.
Reversing location of "kubetest"/namesuffix is a useful change: i.e. from k8s-master-kubetest-routetable to k8s-kubetest-master-routetable
as it makes resource name more readable but I would hold off on this as this also has an impact on some other assumptions.
cc: @JackQuincy for his input
cc: @jackfrancis @seanknox for awareness to not merge this PR without understanding full impact and/or futher discussion.